include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp

```diff
@@ -369,39 +367,37 @@ class two_dimensional_matrix<value_t, allocator_t, order>::iterator_type :
     /*!\name Constructors, destructor and assignment
     * \{
     */
-    constexpr iterator_type() = default; //!< Defaulted.
-    constexpr iterator_type(iterator_type const &) = default; //!< Defaulted.
-    constexpr iterator_type(iterator_type &&) = default; //!< Defaulted.
-    constexpr iterator_type & operator=(iterator_type const &) = default; //!< Defaulted.
-    constexpr iterator_type & operator=(iterator_type &&) = default; //!< Defaulted.
-    ~iterator_type() = default; //!< Defaulted.
+    constexpr basic_iterator() = default; //!< Defaulted.
+    constexpr basic_iterator(basic_iterator const &) = default; //!< Defaulted.
+    constexpr basic_iterator(basic_iterator &&) = default; //!< Defaulted.
```

Comment:
constexpr basic_iterator(**basic_iterator** &&) = default; //!< Defaulted.

This type has also no template arguments.